src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -131,6 +162,128 @@ public class FhirCosmosSearchService : SearchService
             return searchResult;
         }
 
+        private async Task<List<Expression>> PerformChainedSearch(SearchOptions searchOptions, IReadOnlyList<ChainedExpression> chainedExpressions, CancellationToken cancellationToken)
+        {
+            if (!_cosmosConfig.EnableChainedSearch && _requestContextAccessor.FhirRequestContext?.RequestHeaders.TryGetValue(HeaderEnableChainedSearch, out StringValues _) != true)
+            {
+                throw new SearchOperationNotSupportedException(Resources.ChainedExpressionNotSupported);
+            }
+
+            var chainedReferences = new List<Expression>();
+            SearchOptions chainedOptions = searchOptions.Clone();
+            chainedOptions.CountOnly = false;
+            chainedOptions.MaxItemCount = _chainedSearchMaxSubqueryItemLimit;
+            chainedOptions.MaxItemCountSpecifiedByClient = false;
+
+            // Local function for recursion, this lets us walk to the end of the expression to start filtering,
+            // the results are used to filter against the parent layer.
+            async Task<Expression> RecursiveLookup(ChainedExpression expression)
```

Comment:
>async Task<Expression> RecursiveLookup(ChainedExpression expression) [](start = 12, length = 68)

nit: Can it be normal function rather than local? (For better readability?)
It's too big to be local, at least for my taste and size of my monitor. I can't even see where it ends.